src/codegen/doc/mod.rs

```diff
@@ -503,8 +513,42 @@ where
             )?;
         }
 
-        for parameter in fn_.parameters() {
-            if parameter.instance_parameter || parameter.name.is_empty() {
+        // A list of parameters positions to filter out
+        let mut indices_to_ignore: Vec<_> = fn_
+            .parameters()
+            .iter()
+            .filter_map(|param| param.array_length)
+            .collect();
+        if let Some(indice) = fn_.ret().array_length {
+            indices_to_ignore.push(indice);
+        }
+
+        // The original list of parameters without the ones the specifics an array length
+        let no_array_length_params: Vec<_> = fn_
+            .parameters()
+            .iter()
+            .enumerate()
+            .filter_map(|(indice, param)| {
+                if indices_to_ignore.contains(&(indice as u32)) {
+                    None
+                } else {
+                    Some(param)
+                }
```

Comment:
I've been shouted at for suggesting `.then` in another repo, but maybe it works here? `(!indices_to_ignore.contains(&(indice as u32))).then(|| param)`.

Alternatively you might as well separate these out over a `.filter(|(index, _)| ...).map(|(_, param)| param)`.

Reply:
I went with then, the filter & then map isn't great imho :)

Reply:
Yeah, there's no `.enumerate_filter()` or something like that :/

Reply:
OTOH you already have a `filter` over param... :)

Reply:
> OTOH you already have a `filter` over param... :)

I want to keep both separate for understandability otherwise the code becomes cryptic